administration/src/cards/extensions/AddressFieldExtensions.ts

```diff
@@ -1,45 +1,47 @@
+import { CardBlueprint } from '../CardBlueprint'
 import { Extension } from './extensions'
 
-abstract class BaseAddressExtension extends Extension<string, null> {
-  setInitialState() {
-    this.state = ''
-  }
-  causesInfiniteLifetime() {
-    return false
-  }
-  createForm() {
-    return null
-  }
-  isValid(): boolean {
-    return this.state !== null
-  }
-  fromString(state: string): void {
-    this.state = state
-  }
-  toString(): string {
-    return this.state ?? ''
-  }
-}
+export const ADDRESS_LINE_1_EXTENSION = 'addressLine1'
+export const ADDRESS_LINE_2_EXTENSION = 'addressLine1'
+export const ADDRESS_PLZ_EXTENSION = 'addressPlz'
+export const ADDRESS_LOCATION_EXTENSION = 'addressLocation'
 
-class AddressLine1Extension extends BaseAddressExtension {
-  public readonly name = AddressLine1Extension.name
-}
+type AddressFieldExtension =
+  | typeof ADDRESS_LINE_1_EXTENSION
+  | typeof ADDRESS_LINE_2_EXTENSION
+  | typeof ADDRESS_PLZ_EXTENSION
+  | typeof ADDRESS_LOCATION_EXTENSION
+type AddressFieldExtensionState<T extends AddressFieldExtension> = Record<T, string>
 
-class AddressLine2Extension extends BaseAddressExtension {
-  public readonly name = AddressLine2Extension.name
-}
+const getAddressFieldExtension = <T extends AddressFieldExtension>(
+  name: T
+): Extension<AddressFieldExtensionState<T>> => ({
+  name,
+  Component: () => null,
+  getInitialState: () => ({ [name]: '' } as AddressFieldExtensionState<T>),
+  causesInfiniteLifetime: () => false,
+  // TODO Why no protobuf data?
```

Comment:
Is it correct that address fields (and other extensions) are not included in protobuf data?

Reply:
it depends on the extension if its included
```js
  setProtobufData(message: PartialMessage<CardExtensions>): void {
    message.extensionBirthday = {
      birthday: this.state?.birthday,
    }
  }
```

Reply:
Yes, sure. But why is the address not included for example?

Reply:
because its not needed in the qr code. this is just for the pdf letterhead and will not be transferred to the app.